src/specificAnimalClasses/SnakeClass.ts

```diff
@@ -24,6 +24,7 @@ export default class Snake implements ICanWalk, ICanSwim, Reptile {
     color: string,
     length: number
   ) {
+    checkNameNotEmpty(name);
```

Comment:
Why do you need "group" in this class ? What is the use of it, can we somehow make OPP work with us and avoid having a field that tells us what this class inherits ( or implements)